website/addons/dataverse/views.py

```diff
@@ -0,0 +1,332 @@
+)
+
+## Auth ##
+
+@must_be_logged_in
+def dataverse_user_config_get(auth, **kwargs):
+    """View for getting a JSON representation of the logged-in user's
+    Dataverse user settings.
+    """
+
+    user_addon = auth.user.get_addon('dataverse')
+    user_has_auth = False
+    if user_addon:
+        user_has_auth = user_addon.has_auth
+
+    return {
+        'result': {
+            'userHasAuth': user_has_auth,
+            'urls': {
+                'create': api_url_for('dataverse_add_user_account'),
+                'accounts': api_url_for('dataverse_account_list'),
+            },
+            'hosts': DEFAULT_HOSTS,
+        },
+    }, http.OK
+
+
+## Config ##
+
+@must_be_logged_in
+def dataverse_add_user_account(auth, **kwargs):
+    """Verifies new external account credentials and adds to user's list"""
+    user = auth.user
+    provider = DataverseProvider()
+
+    host = request.json.get('host').rstrip('/')
+    api_token = request.json.get('api_token')
+
+    # Verify that credentials are valid
+    client.connect_or_error(host, api_token)
+
+    # Note: `DataverseSerializer` expects display_name to be a URL
+    try:
+        provider.account = ExternalAccount(
+            provider=provider.short_name,
+            provider_name=provider.name,
+            display_name=host,       # no username; show host
+            oauth_key=host,          # hijacked; now host
+            oauth_secret=api_token,  # hijacked; now api_token
+            provider_id=api_token,   # Change to username if Dataverse allows
+        )
+        provider.account.save()
+    except KeyExistsException:
+        # ... or get the old one
+        provider.account = ExternalAccount.find_one(
+            Q('provider', 'eq', provider.short_name) &
+            Q('provider_id', 'eq', api_token)
+        )
+        assert provider.account is not None
+
+    if provider.account not in user.external_accounts:
+        user.external_accounts.append(provider.account)
+
+    user_addon = auth.user.get_addon('dataverse')
+    if not user_addon:
+        user.add_addon('dataverse')
+    user.save()
+
+    # Need to ensure that the user has dataverse enabled at this point
+    user.get_or_add_addon('dataverse', auth=auth)
+    user.save()
+
+    return {}
+
+@must_have_permission('write')
+@must_have_addon(SHORT_NAME, 'user')
+@must_have_addon(SHORT_NAME, 'node')
+@must_be_addon_authorizer(SHORT_NAME)
+def dataverse_set_config(node_addon, auth, **kwargs):
+    """Saves selected Dataverse and dataset to node settings"""
+
+    user_settings = node_addon.user_settings
+    user = auth.user
+
+    if user_settings and user_settings.owner != user:
+        raise HTTPError(http.FORBIDDEN)
+
+    try:
+        assert_clean(request.json)
+    except AssertionError:
+        # TODO: Test me!
+        raise HTTPError(http.NOT_ACCEPTABLE)
+
+    alias = request.json.get('dataverse').get('alias')
+    doi = request.json.get('dataset').get('doi')
+
+    if doi is None:
```

Comment:
`alias` is a required arg for `get_dataverse` below, so checking is probably good.